test/e2e/Tests/Helpers/DurableHelpers.cs

```diff
@@ -31,28 +31,39 @@ public OrchestrationStatusDetails(string statusQueryResponse)
         }
     }
 
-    internal static string ParseStatusQueryGetUri(HttpResponseMessage invocationStartResponse)
+    private static string TokenizeAndGetValueFromKeyAsString(string? json, string key)
     {
-        string? responseString = invocationStartResponse.Content?.ReadAsStringAsync().Result;
-
-        if (string.IsNullOrEmpty(responseString))
+        if (string.IsNullOrEmpty(json))
         {
             return string.Empty;
         }
-        JsonNode? responseJsonNode = JsonNode.Parse(responseString);
+        JsonNode? responseJsonNode = JsonNode.Parse(json);
         if (responseJsonNode == null)
         {
             return string.Empty;
         }
 
-        string? statusQueryGetUri = responseJsonNode["StatusQueryGetUri"]?.GetValue<string>();
+        string? statusQueryGetUri = responseJsonNode[key]?.GetValue<string>();
         return statusQueryGetUri ?? string.Empty;
     }
-    internal static OrchestrationStatusDetails GetRunningOrchestrationDetails(string statusQueryGetUri)
+
+    internal static async Task<string> ParseStatusQueryGetUri(HttpResponseMessage invocationStartResponse)
+    {
+        string? responseString = await invocationStartResponse.Content.ReadAsStringAsync();
+        return TokenizeAndGetValueFromKeyAsString(responseString, "StatusQueryGetUri");
+    }
+
+    internal static async Task<string> ParseInstanceId(HttpResponseMessage invocationStartResponse)
+    {
+        string? responseString = await invocationStartResponse.Content.ReadAsStringAsync();
+        return TokenizeAndGetValueFromKeyAsString(responseString, "Id");
+    }
+
+    internal static async Task<OrchestrationStatusDetails> GetRunningOrchestrationDetails(string statusQueryGetUri)
     {
-        var statusQueryResponse = _httpClient.GetAsync(statusQueryGetUri);
+        var statusQueryResponse = await _httpClient.GetAsync(statusQueryGetUri);
 
-        string? statusQueryResponseString = statusQueryResponse.Result.Content.ReadAsStringAsync().Result;
+        string? statusQueryResponseString = statusQueryResponse.Content.ReadAsStringAsync().Result;
```

Comment:
We should avoid calling `.Result` if at all possible because it can result in deadlocks in certain situations.
```suggestion
        string? statusQueryResponseString = await statusQueryResponse.Content.ReadAsStringAsync();
```

Reply:
Yep - this PR attempted to move all blocking operations to async/await pattern, looks like I missed one. Fixed.